### ISSUE TITLE
$shell is wrong with pyenv-init in a script

### ISSUE THREAD

**Issue by blueyed** [pyenv/pyenv#373]
I've noticed that the $shell setup is wrong, when you're using `eval $(pyenv init -)` in a script:

```
#!/bin/sh
eval "$(pyenv init -)"
echo $PYENV_SHELL
```

It should be "sh", but is the name of the script.

It gets setup in https://github.com/yyuu/pyenv/blob/e285f2e5b1665993f7ebb06eb922bdd97311cbf3/libexec/pyenv-init#L23-30:

```
shell="$1"
if [ -z "$shell" ]; then
  shell="$(ps c -p "$PPID" -o 'ucomm=' 2>/dev/null || true)"
  shell="${shell##-}"
  shell="${shell%% *}"
  shell="$(basename "${shell:-$SHELL}")"
fi
```

The problem is caused by `ps c` / `ucomm`, where `args` should be used instead.

I see that it's an issue with/from rbenv, and will forward it there.
##

**Comment by chrahunt** [comment]
Fixed by #79.

**Comment by blueyed** [comment]
Thanks for closing.
The real fix came with merging rbenv in https://github.com/pyenv/pyenv/pull/549.
